src/weather_gen.cpp

```diff
@@ -42,8 +42,7 @@ w_point weather_generator::get_weather( const tripoint &location, const calendar
     double A( raw_noise_4d( x, y, z, modSEED ) * 8.0 );
     double W;
 
-    const double now( double( t.turn_of_year() + to_turns<int>( calendar::season_length() ) / 2 ) /
-                      to_turns<int>( calendar::year_length() ) ); // [0,1)
+    const double now( time_past_new_year( t ) / calendar::year_length() ); // [0,1)
```

Comment:
@BevapDin: People say weather became somewhat colder (https://discourse.cataclysmdda.org/t/15260) and this line is likely candidate.

Shouldn't it be `const double now( ( time_past_new_year( t ) + calendar::season_length() / 2 ) / calendar::year_length() )`, actually?